pipeline/src/pipeline.rs

```diff
@@ -126,13 +126,19 @@ pub struct Pipeline<T: FieldElement> {
     log_level: Level,
     /// Optional arguments for various stages of the pipeline.
     arguments: Arguments<T>,
+    /// The context for the host.
+    host_context: HostContext,
 }
 
+use super::HostContext;
+
 impl<T> Default for Pipeline<T>
 where
     T: FieldElement,
 {
     fn default() -> Self {
+        let ctx = HostContext::new();
```

Comment:
i was thinking, maybe the constructor could already return `(HostContext, QueryCallback)`, one doesn't make sense without the other, and the `query_callback` method won't ever be called more than once, i think.